crypto/key.go

```diff
@@ -345,7 +344,9 @@ func KeyEqual(k1, k2 Key) bool {
 		return true
 	}
 
-	b1, err1 := k1.Bytes()
-	b2, err2 := k2.Bytes()
-	return bytes.Equal(b1, b2) && err1 == err2
+	if k1.Type() != k2.Type() {
+		return false
+	}
```

Comment:
Let's move the type check into the `Equals` implementation. I believe we call that manually in some places.